src/wordcount.js

```diff
@@ -229,9 +230,19 @@ export default class WordCount extends Plugin {
 	_calculateWordsAndCharacters() {
 		const txt = modelElementToPlainText( this.editor.model.document.getRoot() );
 
-		this.characters = txt.length;
+		this.characters = txt.replace( /\n/g, '' ).length;
 
-		this.words = ( txt.match( /[_a-zA-Z0-9À-ž]+/gu ) || [] ).length;
+		// Groups:
+		// {L} - Any kind of letter from any language.
+		// {N} - Any kind of numeric character in any script.
+		// {M} - A character intended to be combined with another character (e.g. accents, umlauts, enclosing boxes, etc.).
+		// {Pd} - Any kind of hyphen or dash.
+		// {Pc} - A punctuation character such as an underscore that connects words.
+		const wordsMatch = regExpFeatureDetection.isUnicodePropertySupported ?
+			txt.match( new RegExp( '[\\p{L}\\p{N}\\p{M}\\p{Pd}\\p{Pc}]+', 'gu' ) ) :
+			txt.match( /[_\-a-zA-Z0-9À-ž]+/gu );
```

Comment:
Sorry I've missed that earlier...

I got mixed feelings about this ternary - there are two types of RegExp creation where one would be sufficient as we do not create pattern on the fly we could use the latter `/foo/` and not the constructor one but either one will be OK.

The second thing is that then we could move the regexp choosing to a variable:

```js
const wordsMatch = txt.match( matchRegExp ) || []; // also \\ [] can be moved here.
```

Reply:
ps.: I've just found out that ESLint is not allowing Unicode groups in `//` form. So we must string pattern and `new RegExp()`.

Reply:
> ps.: I've just found out that ESLint is not allowing Unicode groups in // form. So we must string pattern and new RegExp().

Yeah that's exactly the cases why it's used in this form. Maybe there should be a comment about it, as it isn't clear why we use this form?

Reply:
It appears that after reinstall dependencies, the ESLint doesn't mark this literal as invalid expression. So I change it.